gst_msdk/gst-libs/mfx/gstmfxfilter.c

```diff
@@ -888,14 +888,20 @@ gst_mfx_filter_set_deinterlace_mode (GstMfxFilter * filter,
   mfxExtVPPDeinterlacing *ext_deinterlacing;
 
   g_return_val_if_fail (filter != NULL, FALSE);
+
+#if MSDK_CHECK_VERSION(1,19)
```

Comment:
I think your change here does not mean the same thing as the original code. Indeed, the original code has compilation issues with g_return_val_if_fail() unless I define the G_DISABLE_CHECKS macro. I suggest to make it simpler we can just stick to a classic if (condition_not_met) return FALSE, instead of using g_return_val_fail(). For example:

  if (GST_MFX_DEINTERLACE_MODE_BOB != mode
      || GST_MFX_DEINTERLACE_MODE_ADVANCED != mode
      || GST_MFX_DEINTERLACE_MODE_ADVANCED_NOREF != mode
#if MSDK_CHECK_VERSION(1,19)
      || GST_MFX_DEINTERLACE_MODE_ADVANCED_SCD != mode
      || GST_MFX_DEINTERLACE_MODE_FIELD_WEAVING != mode
#endif
    )
    return FALSE;